crypto/rsa_extra/rsa_test.cc

```diff
@@ -578,13 +578,71 @@ TEST(RSATest, OnlyDGiven) {
   ASSERT_TRUE(BN_hex2bn(&key2->d, kD));
   key2->flags |= RSA_FLAG_NO_BLINDING;
 
+  // While keys defined only in terms of |n| and |d| must be functional, our
+  // validation logic doesn't consider them "valid".
+  EXPECT_FALSE(RSA_check_key(key2.get()));
+
   ASSERT_LE(RSA_size(key2.get()), sizeof(buf));
   EXPECT_TRUE(RSA_sign(NID_sha256, kDummyHash, sizeof(kDummyHash), buf,
                        &buf_len, key2.get()));
 
   // Verify the signature with |key|. |key2| has no public exponent.
   EXPECT_TRUE(RSA_verify(NID_sha256, kDummyHash, sizeof(kDummyHash), buf,
                          buf_len, key.get()));
+
+  // Perform the same test case as above ("ACCP-style" key specified in terms
+  // of |d| and |n|, turn off blinding), but this time do a round trip to DER +
+  // ASN.1 first. This most closely replicates ACCP's use-case of asking AWS-LC
+  // to encode to and decode from DER. While already-parsed keys tend to be
+  // validated and used with "NULL"ness indicating absence of various RSA
+  // parameters, the actual parsing logic assumes that these NULL'd values are
+  // present and zero-valued when being un/marshalled. So, we zero them out
+  // before marshalling, and assert that they were NULLed when parsing back
+  // from DER.
+  //
+  // Also, note that here and in the previous test we expect RSA_check_key to
+  // return false for ACCP-style keys. The current implementation does not
+  // consider ACCP-style keys to be valid. We deliberately made this decision
+  // to avoid introducing potentially dangerous modifications to key validation
+  // and generation logic. Instead, we detect ACCP-style keys only when parsing
```

Comment:
```suggestion
  // and generation logic. Instead, we detect JCA-style keys only when parsing
```

Reply:
i have no issue with that. perhaps we can consolidate discussion with @SalusaSecondus around the naming issue on [this thread](https://github.com/awslabs/aws-lc/pull/436#discussion_r850827317)?